test/spec/BuildSpec.hs

```diff
@@ -11,105 +12,166 @@ import Test.Mockery.Directory
-
-        it "complains about packages that cannot be built" $ \onTestFailureLog -> do
-          writeHaskellProject repoDir
+    wrap $ do
+      it "builds packages and creates a result link" $ \runGarn -> do
+        writeHaskellProject repoDir
+        _ <- runGarn ["build", "foo"]
+        doesDirectoryExist "result" `shouldReturn` True
+        StdoutTrim output <- cmd ("result/bin/garn-test" :: String)
+        output `shouldBe` ("haskell test output" :: String)
+
+      it "complains about packages that cannot be built" $ \runGarn -> do
+        writeHaskellProject repoDir
+        writeFile
+          "Main.hs"
+          [i|
+            main :: IO ()
+            main = "foo"
+          |]
+        output <- runGarn ["build", "foo"]
+        stderr output `shouldContain` "Couldn't match type"
+        exitCode output `shouldBe` ExitFailure 1
+
+      it "allows to specify which sub-package to build" $ \runGarn -> do
+        writeFile "garn.ts" $
+          unindent
+            [i|
+              import * as garn from "#{repoDir}/ts/mod.ts";
+
+              export const project = garn.mkProject(
+                { description: "" },
+                {
+                  foo: garn.build`echo 'foo' > $out/build-artifact`,
+                  bar: garn.build`echo 'bar' > $out/build-artifact`,
+                },
+              );
+            |]
+        output <- runGarn ["build", "project.foo"]
+        exitCode output `shouldBe` ExitSuccess
+        readFile "result/build-artifact" `shouldReturn` "foo\n"
+        output <- runGarn ["build", "project.bar"]
+        exitCode output `shouldBe` ExitSuccess
+        readFile "result/build-artifact" `shouldReturn` "bar\n"
+
+      it "builds top-level packages" $ \runGarn -> do
+        writeFile "garn.ts" $
+          unindent
+            [i|
+              import * as garn from "#{repoDir}/ts/mod.ts";
+
+              export const p = garn.build`echo 'build output' > $out/build-artifact`;
+            |]
+        output <- runGarn ["build", "p"]
+        exitCode output `shouldBe` ExitSuccess
+        readFile "result/build-artifact" `shouldReturn` "build output\n"
+
+      it "shows a nice help for packages" $ \runGarn -> do
+        writeFile "garn.ts" $
+          unindent
+            [i|
+              import * as garn from "#{repoDir}/ts/mod.ts";
+
+              export const project = garn.mkProject(
+                { description: "" },
+                {
+                  short: garn.build`short command`,
+                  longer: garn.build`
+                    # this is some longer build script:
+                    bla bla bla
+                  `,
+                },
+              );
+            |]
+        output <- runGarn ["build"]
+        stderr output `shouldContain` "project.short            Builds short command"
+        stderr output `shouldContain` "project.longer           Builds # this is some longe..."
```

Comment:
TIOLI, just felt like a change of number of spaces between the name and description shouldn't break this test
```suggestion
        stderr output `shouldMatch` "project.short \\s+ Builds short command"
        stderr output `shouldMatch` "project.longer \\s+ Builds # this is some longe\\.\\.\\."
```

(The extra spaces around `\\s+` make it a bit easier to read, and will only fail if there are fewer than 3 spaces which felt reasonable to me)